pkg/client/fake/client.go

```diff
@@ -114,11 +115,20 @@ func (t versionedTracker) Update(gvr schema.GroupVersionResource, obj runtime.Ob
 			field.ErrorList{field.Required(field.NewPath("metadata.name"), "name is required")})
 	}
 
+	gvk := obj.GetObjectKind().GroupVersionKind()
+	emptyGvk := schema.GroupVersionKind{}
+	if gvk == emptyGvk {
```

Comment:
```suggestion
	if gvk == schema.GroupVersionKind{} {
```

Reply:
@vincepri I have tested this syntax first, but it is failing to compile with "expected declaration, found oldObject"

Reply:
Oh I think you can just do `gvk.Empty()`